src/index.js

```diff
@@ -48,5 +44,7 @@ const plugin = {
 	},
 };
 
+Object.assign(plugin.configs.recommended.plugins, { json: plugin });
```

Comment:
Can we simplify like this?

```suggestion
plugin.configs.recommended.plugins.json =plugin;
```

Reply:
I had to put the assignment in braces so that TypeScript would still infer the type correctly. If the braces are removed, the generated type for `plugin` will look just like this:

```ts
declare namespace plugin {
    namespace configs {
        namespace recommended {
            namespace plugins {
                export { plugin as json };
            }
        }
    }
}
```

Another solution that works is putting the assignment in parentheses and disabling prettier inline to ensure that the parentheses are not removed.

My feeling is that TypeScript's type inference is not predictable enough to be trusted without type tests because a seemingly small change can break the types, and that could happen inadvertently.

Reply:
I opened a change request to add type tests: #61.